src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -21,6 +21,32 @@ public abstract class BaseSymbolicLinks_FileSystem : BaseSymbolicLinks
         /// <summary>Calls the actual public API for resolving the symbolic link target.</summary>
         protected abstract FileSystemInfo ResolveLinkTarget(string linkPath, string? expectedLinkTarget, bool returnFinalTarget = false);
 
+        /// <summary>
+        /// Appends \??\ to the absolute path
+        /// </summary>
+        protected void AssertPathEquals(string expected, string actual)
+        {
+#if WINDOWS // Windows implementation has a couple nuances compared to Unix.
+            // If the link target path is absolute, the following occurs:
+            // ResolveLinkTarget(), which uses DeviceIoControl, will return the path with the DOS Device prefix (\??\).
+            // ResolveLinkTarget(returnFinalTarget: true), which uses GetFinalPathNameByHandleW, will return with the extended prefix (\\?\).
+            // For testing, we will ignore this prefix.
+            if (PathInternal.IsExtended(actual))
+            {
+                actual = actual.Substring(4);
+            }
+
+            // Windows syscalls remove the redundant segments in the link target path.
+            // We will remove them from the expected path when testing Windows but keep them when testing Unix, which doesn't removes them.
```

Comment:
It seems that `FileSystemInfo.FullName` calls `RemoveRelativeSegments`. We'll see if this needs to be adjusted once it hits the CI.

BTW, typo:

```suggestion
            // We will remove them from the expected path when testing Windows but keep them when testing Unix, which doesn't remove them.
```

Reply:
the `expected` path doesn't come from `FullName`.  I think in all case comes from `GetRandomLinkPath()`.